packages/validate-icu-locales/src/index.ts

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env node
+
+import { parse } from '@formatjs/icu-messageformat-parser'
+import { info } from 'console'
+import { readFile } from 'fs/promises'
+import { globby } from 'globby'
+import { importFromString } from 'module-from-string'
+
+const args = process.argv.slice(2)
+const pattern = args[0]
+const { error, table } = console
+
+let isICUError = false
+
+const findICUError = (locales: { [key: string]: string }, filePath: string) => {
+  Object.keys(locales).forEach(key => {
+    const value = locales[key]
+
+    try {
+      parse(value)
+    } catch (err) {
+      isICUError = true
+      error({
+        err,
+        value,
+        key,
+        filePath,
+      })
+    }
+  })
+}
```

Comment:
I don't understand why you are trying to complicate things 😄 

Why not return the error and combine them in the readFiles ? Then if the readFiles return an array it means there is error